Bug: eslint-plugin@6.1.0/rules.d.ts contains TS error

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

When installing `@typescript-eslint/eslint-plugin@6.1.0` and importing it into a TS file (which causes it to be type-checked), `@typescript-eslint/eslint-plugin/rules.d.ts` throws the following TypeScript error (with the latest version of TypeScript, 5.1.6):

```
node_modules/@typescript-eslint/eslint-plugin/rules.d.ts:44:1 - error TS2309: An export assignment cannot be used in a module with other exported elements.
44 export = rules;
```

This is using the following `tsconfig.json`:
```
{
	"compilerOptions": {
		"allowSyntheticDefaultImports": true,
		"moduleResolution": "nodenext",
		"target": "esnext"
	}
}
```

### Reproduction Repository Link

https://github.com/typescript-eslint/typescript-eslint/files/12080240/repro.zip

### Repro Steps

1) download and extract [repro.zip](https://github.com/typescript-eslint/typescript-eslint/files/12080240/repro.zip)
2) `npm i`
3) `npx tsc`


### Versions

<!-- Fill in the relevant versions and delete the others -->
| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.1.0` |
| `@typescript-eslint/parser`            | `6.1.0` |
| `@typescript-eslint/scope-manager`     | `6.1.0` |
| `@typescript-eslint/typescript-estree` | `6.1.0` |
| `@typescript-eslint/type-utils`        | `6.1.0` |
| `@typescript-eslint/utils`             | `6.1.0` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.45.0` |
| `node`                                 | `20.1.0` |

--- a/packages/eslint-plugin/rules.d.ts
+++ b/packages/eslint-plugin/rules.d.ts
@@ -41,4 +41,4 @@ export interface TypeScriptESLintRules {
   [ruleName: string]: RuleModule<string, unknown[]>;
 }
 declare const rules: TypeScriptESLintRules;
-export = rules;
+export default rules;

